src/js/common.js

```diff
@@ -1,12 +1,7 @@
-function trace(text) {
-  // This function is used for logging.
-  if (text[text.length - 1] === '\n') {
-    text = text.substring(0, text.length - 1);
-  }
-  if (window.performance) {
-    var now = (window.performance.now() / 1000).toFixed(3);
-    console.log(now + ': ' + text);
-  } else {
-    console.log(text);
-  }
+ /* exported trace */
+
+// Logging utility function.
+function trace(arg) {
+  var now = (window.performance.now() / 1000).toFixed(3);
+  console.log(now + ': ', arg);
```

Comment:
Are we likely to encounter objects without a stringifier in the this context?

> removing trailing NL

Yes :). I removed this, because it adds complexity and removing it doesn't seem to cause much trouble.